### PR TITLE
[FEATURE] Convert Existing Self-Initializing Expectations to Make ExpectationConfigurationBuilder Self-Contained with its own validation_parameter_builder settings

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* Make `ExpectationConfigurationBuilder` component of default `RuleBasedProfilerConfig` in existing Self-Initializing Expectations use its `validation_parameter_builder_configs` settings, making it self-contained, and thus portable for use as a building block in creating multi-expectation profilers.
* Use formally defined syntax and constants in existing Self-Initializing Expectations to facilitate general developer experiences of creating multi-expectation profilers using the `RuleBasedProfilerConfig` in Self-Initializing Expectations as ready-made building block templates.
* Update existing unit tests and add new unit tests, which illustrate the flexibility of specifying `ParameterBuilder` configurations as part of its own `parameter_builders` layer, and/or as part of `evaluation_parameter_builder_configs` property of any `ParameterBuilder` and/or as part of `validation_parameter_builder_configs` property of an `ExpectationConfigurationBuilder` -- any mix works properly.
* Remove unnecessary `value_range` suffix from output fully-qualified parameter name of `NumericMetricRangeMultiBatchParameterBuilder` and update existing fixtures, tests, notebook(s) and documentation.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-464/GREAT-688/GREAT-695
-
-

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/test_fixtures/rule_based_profiler/example_notebooks/BasicExample_RBP_Instantiation_and_running.ipynb [hunk 1]
```diff
@@ -1014,8 +1014,8 @@
     "expect_column_max: DefaultExpectationConfigurationBuilder = DefaultExpectationConfigurationBuilder(\n",
     "    expectation_type=\"expect_column_max_to_be_between\",\n",
     "    column=\"$domain.domain_kwargs.column\", \n",
-    "    min_value=\"$parameter.max_range_parameter_builder.value.value_range[0]\",\n",
-    "    max_value=\"$parameter.max_range_parameter_builder.value.value_range[1]\",\n",
+    "    min_value=\"$parameter.max_range_parameter_builder.value[0]\",\n",
+    "    max_value=\"$parameter.max_range_parameter_builder.value[1]\",\n",
     ")"
```
Comment (Shinnnyshinshin) [inline tests/test_fixtures/rule_based_profiler/example_notebooks/BasicExample_RBP_Instantiation_and_running.ipynb:1019]: Thank you for this @alexsherstinsky .. it really makes simpler and more elegant. I love it

File: great_expectations/expectations/core/expect_column_max_to_be_between.py [hunk 2]
```diff
@@ -104,6 +115,30 @@ class ExpectColumnMaxToBeBetween(ColumnExpectation):
         "profiler_config",
     )
 
+    max_range_estimator_parameter_builder_config: ParameterBuilderConfig = (
+        ParameterBuilderConfig(
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/core/expect_column_max_to_be_between.py:119]: Just confirming: this is the pattern going forward for self-initializing Expectations right? I remember you mentioned it during our Standup today

**Review by Shinnnyshinshin — APPROVED** [review]
Wonderful work @alexsherstinsky. Thank you for this :)

File: great_expectations/expectations/core/expect_column_max_to_be_between.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/expectations/core/expect_column_max_to_be_between.py:119]: @Shinnnyshinshin This pattern will make things easier; however, the more verbose pattern is available as well.  I think there are four ways to pick from as situation calls for:
* Using `ParameterBuilderConfig` (as in above)
* Using `dict` version of the above
* Using `ParameterBuilderConfig.to_dict()` -- if starting with `ParameterBuilderConfig` and wanting a plain dictionary.
* Using the specific `ParameterBuilder` object and methods on it and/or `ParameterBuilder.config()` and methods on it and/or `ParameterBuilder.to_json_dict()` and keys from it.

Any and all these are supported in any combination.  For Self-Initializing Expectations, I would use the first or the fourth approaches.

Thanks!

**Pull request merged**